tests/system/Debug/Toolbar/Collectors/HistoryTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Debug\Toolbar\Collectors;
+
+use CodeIgniter\Test\CIUnitTestCase;
+use DateTime;
+
+/**
+ * @internal
+ */
+final class HistoryTest extends CIUnitTestCase
+{
+    private const STEP = 0.0001;
+
+    protected float $time;
```

Comment:
private float $time;

Reply:
Why is that important because, this is final class, also this is test class.

Reply:
Exactly. This is a final class. You use `protected` because you intend child classes to be able to use the property. However, this is already final so there will be no inheritance. Protected visibility is pointless.